src/Common/Acl/AclExtended.php

```diff
@@ -236,9 +237,12 @@ public static function setUserAro($arr_group_titles, $user_name, $first_name, $m
         //see if this user is gacl protected (ie. do not allow
         //removal from the Administrators group)
         require_once(dirname(__FILE__) . '/../../../library/user.inc');
-        require_once(dirname(__FILE__) . '/../../../library/calendar.inc');
-        $userNametoID = getIDfromUser($user_name);
-        if (checkUserSetting("gacl_protect", "1", $userNametoID) || $user_name == "admin") {
+
+        $userService = new UserService();
+        $user = $userService->getIdByUsername($user_name);
+        $userNameToID = $user['id'];
```

Comment:
with below rec on function, then this will become:
```
$userNameToID = $userService->getIdByUsername($user_name);
```
since only using this class instance to do one thing here, could also make it a one liner (I only do this when I am confident that the class instance will only be used one time in the script):
```
$userNameToID = (new UserService())->getIdByUsername($user_name);
```